### DISCUSSION TITLE
Using CoreDNS to replace my authoritative nameserver

### DISCUSSION THREAD

**Discussion by dploeger** [coredns/coredns#5870] [Q&A]
Hey there!

I'm currently building up a replacement for my old hosting service - completely based on Kubernetes.

In my old setup I was using a manually configured Bind that served as my primary authoritative nameserver with two secondaries using a free service.

In the new setup I wanted to use CoreDNS as a replacement for Bind. There are two parts in this: The dynamic part using external-dns which stores ingress resource lookup in an etcd and a static part for all the usual stuff like TXT records or specific vanity CNAMEs etc.

I tried various things, but I can't seem to get the two combined. Only using etcd enables a dynamic SOA and A-records when using external-dns, but limited resource types. If I add the file plugin, I need to specify a SOA record with a serial that is not managed by the etcd plugin when a dynamic RR changes.

One possibility I see would be to use the external plugin "records" which apparently doesn't require a SOA record, but that would need another container image than the official one.

Also, the dynamic SOA record is pretty fixed to a static RNAME and MNAME, but I could work around that. And I need to test the transfer plugin to enable connecting the secondaries.

Any hints on how to achieve a dynamic setup based on Kubernetes ingress with just some static records injected from somewhere?

Thanks!

Kind regards 
Dennis

**Comment by chrisohaver** [comment]
> .. I was using a manually configured Bind that served as my primary authoritative nameserver with two secondaries ...
> In the new setup I wanted to use CoreDNS as a replacement for Bind.

This sounds straightforward. CoreDNS can replace those functions with just the _file_ and _transfer_ plugins.

I guess the "dynamic part" is not something your BIND solution was doing, and perhaps your driver for moving to CoreDNS from BIND?

FWIW, the _transfer_ plugin for the most part does not work in conjunction with dynamic records.  Not all plugins support transfers, and the _transfer_ plugin only collects records from a single plugin at a time. For example,  if you used the _file_ plugin to serve zone `example.com`, and used the _template_ plugin to serve some dynamic records for the same zone, a transfer for `example.com` would only pick up the records defined by the _file_ plugin, and none of the record produced by _template_.

**Reply by dploeger** [reply]
Thank you. Can you tell me how the dynamic part can be done using the file plugin? I can see it supporting the static records (like i.e. google verification TXT records) but not records that come from Kubernetes ingress resources (as provided by external-dns).

External-dns supports RFC2136 dynamic records but as far as I understood, CoreDNS hasn't implemented that yet?

**Reply by chrisohaver** [reply]
> Thank you. Can you tell me how the dynamic part can be done using the file plugin?

The file plugin does not create dynamic records.
